tools/bench.md

```diff
@@ -0,0 +1,222 @@
+
+```ts
+// Do 100k warmup runs and 1 million measured runs
+Deno.bench({ warmup: 1e5, n: 1e6 }, function resolveUrl() {
+  new URL("./foo.js", import.meta.url);
+});
+```
+
+## Running benchmarks
+
+To run a benchmark, call `deno bench` with the file that contains your bench
+function. You can also omit the file name, in which case all benchmarks in the
+current directory (recursively) that match the glob
+`{*_,*.,}bench.{ts, tsx, mts, js, mjs, jsx, cjs, cts}` will be run. If you pass
+a directory, all files in the directory that match this glob will be run.
+
+The glob expands to:
+
+- files named `bench.{ts, tsx, mts, js, mjs, jsx, cjs, cts}`,
+- or files ending with `.bench.{ts, tsx, mts, js, mjs, jsx, cjs, cts}`,
+- or files ending with `_bench.{ts, tsx, mts, js, mjs, jsx, cjs, cts}`
+
+```shell
+# Run all benches in the current directory and all sub-directories
+deno bench
+
+# Run all benches in the util directory
+deno bench util/
+
+# Run just my_bench.ts
+deno bench my_bench.ts
+```
+
+> ⚠️ If you want to pass additional CLI arguments to the bench files use `--` to
+> inform Deno that remaining arguments are scripts arguments.
+
+```shell
+# Pass additional arguments to the bench file
+deno bench my_test.ts -- -e --foo --bar
+```
+
+`deno bench` uses the same permission model as `deno run` and therefore will
+require, for example, `--allow-write` to write to the file system during
+benching.
+
+To see all runtime options with `deno bench`, you can reference the command line
+help:
+
+```shell
+deno help bench
+```
+
+## Filtering
+
+There are a number of options to filter the benches you are running.
+
+### Command line filtering
+
+Benches can be run individually or in groups using the command line `--filter`
+option.
+
+The filter flags accept a string or a pattern as value.
+
+Assuming the following benches:
+
+```ts
+Deno.bench({ name: "my-bench", fn: myBench });
+Deno.bench({ name: "bench-1", fn: bench1 });
+Deno.bench({ name: "bench2", fn: bench2 });
+```
+
+This command will run all of these benches because they all contain the word
+"bench".
+
+```shell
+deno bench --filter "bench" benchmarks/
+```
+
+On the flip side, the following command uses a pattern and will run the second
+and third benchmarks.
+
+```shell
+deno bench --filter "/bench-*\d/" benchmarks/
+```
+
+_To let Deno know that you want to use a pattern, wrap your filter with
+forward-slashes like the JavaScript syntactic sugar for a REGEX._
+
+### Bench definition filtering
+
+Within the benches themselves, you have two options for filtering.
+
+#### Filtering out (ignoring these benches)
+
+Sometimes you want to ignore benches based on some sort of condition (for
+example you only want a benchmark to run on Windows). For this you can use the
+`ignore` boolean in the bench definition. If it is set to true the test will be
+skipped.
+
+```ts
+Deno.bench({
+  name: "do macOS feature",
+  ignore: Deno.build.os !== "darwin",
```

Comment:
The above says "for example you only want a benchmark to run on Window". We should align these two. Also, I think this was meant to be `=== "darwin"`.